easybuild/easyblocks/b/binutils.py

```diff
@@ -78,10 +78,15 @@ def configure_step(self):
 
             libs += ' '.join('-Wl,-rpath=%s' % libdir for libdir in libdirs)
 
-        # statically link to zlib if it is a (build) dependency
+        # configure using `--with-system-zlib` if zlib is a (build) dependency
         zlibroot = get_software_root('zlib')
         if zlibroot:
             self.cfg.update('configopts', '--with-system-zlib')
+
+        # statically link to zlib only if it is a build dependency
+        # see https://github.com/easybuilders/easybuild-easyblocks/issues/1350
+        build_deps = self.cfg.dependencies(build_only=True)
+        if any(dep['name'] == 'zlib' for dep in build_deps):
             libz_path = os.path.join(zlibroot, get_software_libdir('zlib'), 'libz.a')
```

Comment:
@geimer This should be intended one level more, there's no point in checking whether `zlib` was a build-only dep if `zlibroot` is not defined (since `get_software_libdir` will crash hard then)

Reply:
It won't crash, as there isn't a way to list `zlib` as a build dependency without `zlibroot` being defined. But I agree, it could be indented.